encryption_formats.md

```diff
@@ -304,10 +359,123 @@ The IV seed is randomly generated for each chunk.
 
-This designs avoids using the same Key+IV combination multiple times if we need to support chunk rewriting, while also protecting against a malicious storage service that tries to reorder the chunks.
+This design avoids using the same Key+IV combination multiple times if we need to support chunk rewriting, while also protecting against a malicious storage service that tries to reorder the chunks.
 
 The Key is not derived and stays the same for all the chunks.
 
 #### Usage
 
-Currently used to encrypt clear data with byte length >= 1 MB (e.g. big files).
+Currently used to encrypt clear data with byte length >= 1 MB (e.g. big files) when using manual encryption sessions.
+
+### Encryption format v11
+
+#### Spec
+
+This format aims at providing stream encryption by splitting the data into chunks. The maximum encrypted chunk size is encoded in the header and is not supposed to always be the same (our current default: 1MB).
+
+Here is the format of an encrypted chunk:
+
+| **Element**          | **Buffer type**      | **Byte length**    |
+|----------------------|----------------------|--------------------|
+| Version number (11)  | Fixed length         | 1 byte             |
+| Session ID           | Fixed length         | 16 bytes           |
+| Resource ID / Seed   | Fixed length         | 16 bytes           |
+| Encrypted chunk size | Uint32 little endian | 4 bytes            |
+| Encrypted chunk data | Variable length      | = clear chunk size |
+| Encrypted chunk MAC  | Fixed length         | 16 bytes           |
+
+All encrypted chunks have the same size (= maximum encrypted chunk size), except the last one which has always a shorter size.
+
+#### Properties
+
+Total clear data size: TCDS (variable)
+
+Header overhead: HO = 37 bytes
+
+Overhead per chunk: CO = 16 bytes
+
+Encrypted chunk size: ECS = variable (default value: 1 MB)
+
+Clear chunk size: CCS = ECS - CO (default value: 1 MB - 16 bytes)
+
+Number of chunks: NC = ceil[ (TCDS + 1) / CCS ]
+
+Total variable overhead: O = HO + CO * NC
+
+
+
+When the total **clear** data size is a multiple of the maximum **clear** chunk size, an additional “empty” chunk is added at the end. This ensures that the encrypted data is not truncated by a malicious storage service. Corresponding formula:
+TCDS % (ECS - CO) = 0  =>  +1 “empty” chunk of length CO = 16 bytes
+
+The Resource ID is randomly generated once, and is also the seed used to derive the individual encryption key of this resource.
+
+The IV used in encryption is derived from the session ID and the index of the chunk with the following formula: IV = H(session ID, index). The index is concatenated as an uint64 little endian and starts from 0.
+
+This design avoids using the same Key+IV combination multiple times if we need to support chunk rewriting, while also protecting against a malicious storage service that tries to reorder the chunks.
```

Comment:
It does not protect in case of rewriting (we don't have a random seed be chunk anymore) 

```suggestion
This design avoids using the same Key+IV combination multiple times while also protecting against a malicious storage service that tries to reorder the chunks.
```